### PR TITLE
perf(solution): add 'add sso' in execute user task

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts [hunk 1]
```diff
@@ -704,3 +708,133 @@ export type ParamForRegisterTeamsAppAndAad = {
...
+      name: TeamsAppSolutionNameV2,
+      version: "1.0.0",
+      hostType: "Azure",
+      capabilities: [],
+      azureResources: [],
+      activeResourcePlugins: [],
+    };
+    ctx.projectSetting.solutionSettings = solutionSettings;
+    //aad need this file
+    await fs.writeJSON(`${inputs.projectPath}/permissions.json`, DEFAULT_PERMISSION_REQUEST, {
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:789]: This file should not need when there is a aad.template.json file

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts [hunk 2]
```diff
@@ -704,3 +708,133 @@ export type ParamForRegisterTeamsAppAndAad = {
...
+      spaces: 4,
+    });
+  }
+
+  // Check whether can add sso
+  const canProceed = canAddSso(solutionSettings, ctx.telemetryReporter);
```
Comment (SLdragon) [inline packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:795]: Do we need to put this check at the beginning?

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts (hunk 1 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:789]: Good catch. Will update this, thx！

**Review by jayzhang — APPROVED** [review]

**Review by SLdragon — DISMISSED** [review]

File: packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts (hunk 2 above)
Comment (1yefuwang1) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts:807]: It seems having `canAddSso()` here can cause dangling AddSsoStart  event without a subsequent AddSso success/failure event due to early return.

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**